lib/workers/pr/index.ts

```diff
@@ -125,11 +125,19 @@ export function getPlatformPrOptions(
       config.gitLabAutomerge,
   };
 }
-export type EnsurePrResult = {
-  prResult: PrResult;
-  pr?: Pr;
+
+export type ResultWithPr = {
+  pr: Pr;
+  prBlockedBy?: never;
+};
+
+export type ResultWithoutPr = {
+  pr?: never;
+  prBlockedBy: PrBlockedBy;
 };
```

Comment:
Use interface where possible.

@JamieMagee do you know a better typing than using optional never property?

Reply:
Is this not a good place for a discriminated union? Add a `type` or `kind` property and use that to filter on later: https://basarat.gitbook.io/typescript/type-system/discriminated-unions

Reply:
Essentially the truthiness of `pr` is the discrimination field, while the `never` fields are extra precautions to prevent misuse.